File: src/main/java/com/bluehabit/budgetku/common/PermissionUtils.java
/*
 * Copyright © 2023 Blue Habit.
 *
 * Unauthorized copying, publishing of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 */

package com.bluehabit.budgetku.common;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.Arrays;
import java.util.HashSet;

public class PermissionUtils {
    public static boolean hasAccess(String... permissions) {
        var authority = SecurityContextHolder
                .getContext()
                .getAuthentication()
                .getAuthorities()
                .stream()
                .map(GrantedAuthority::getAuthority)
                .toList();

        return new HashSet<>(authority).containsAll(Arrays.stream(permissions).toList());
    }
}

File: src/main/kotlin/com/bluehabit/budgetku/common/utils/PermissionUtils.kt
/*
 * Copyright © 2023 Blue Habit.
 *
 * Unauthorized copying, publishing of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 */

package com.bluehabit.budgetku.common.utils

import com.bluehabit.budgetku.data.role.permission.Permission
import org.springframework.security.core.GrantedAuthority


@JvmName("allowToPermission")
fun Collection<Permission>.allowTo(vararg permission:String): Boolean {
    return this.map { it.permissionType }.containsAll(permission.toList())
}

fun Collection<GrantedAuthority>.allowTo(vararg permission:String): Boolean {
    return this.map { it.authority }.containsAll(permission.toList())
}